src/uri/uri.cc

```diff
@@ -249,6 +249,29 @@ auto URI::path() const -> std::optional<std::string> {
   return path_;
 }
 
+auto URI::path(std::string_view path) -> URI & {
+  if (path.empty()) {
+    this->path_ = std::nullopt;
+    return *this;
+  }
+
+  const auto query_pos = path.find('?');
```

Comment:
This logic makes me think you might want to parse the `path` as a proper `URI`, and then move its `path` member to the class?

Reply:
Is that approach not "too much"? In term of performance :)

Reply:
I mean I really like it, principaly because it will simplify the code ;)

Reply:
The thing is that you are parsing anyway to some extent, and that "some extent" might not cover all edge cases you want to cover, so if we are already doing some sort of parsing, maybe we do all of it?

In any case, let's make it works first, elegantly, covering all edge cases, etc and then we can profile & optimise later on if needed